src/main/java/com/epam/edumanagementsystem/admin/timetable/rest/service/impl/CoursesForTimetableServiceImpl.java

```diff
@@ -100,4 +110,12 @@ public List<String> getCoursesNamesByDayOfWeekAndStatusAndAcademicClassId(String
         return namesOfCourses;
     }
 
+    @Override
+    public List<String> getDoneCoursesNamesByDayOfWeekAndAcademicClassId(String dayOfWeek, Long academicClassId) {
+        List<String> namesOfCourses = new ArrayList<>();
+        doneCoursesService.findAllByAcademicClassId(academicClassId).stream()
+                .filter(doneCourse -> doneCourse.getDate().getDayOfWeek().toString().equalsIgnoreCase(dayOfWeek))
+                .forEach(doneCourse -> namesOfCourses.add(doneCourse.getAcademicCourse().getName()));
```

Comment:
Instead of declaring a list and adding to it, first use map function to get the academic course names and then use collect method

Reply:
Done you can review.